scripts/inference.py

```diff
@@ -0,0 +1,117 @@
+import argparse
+import os
+
+import pandas as pd
+import torch
+from datasets import Dataset, load_dataset
+from jaxtyping import Int
+from tqdm.auto import tqdm
+from transformers import AutoModelForCausalLM
+
+from delphi.eval.utils import get_all_and_next_logprobs, load_validation_dataset
+
+torch.set_grad_enabled(False)
+
+
+def main(
+    model_name: str,
+    batch_size: Int,
+    dataset_name: str,
+    username: str,
+    token: str,
+    funct_test: bool = False,
+):
+    """
+    Outputs the log probabilities of the next token for each token in the validation dataset.
+    And uploads the resulting dataset to huggingface.
+    Args:
+    - model_name: The name of the model to use for inference
+    - batch_size: The batch size for processing. 80 worked well in CPU.
+    - dataset_name: The name of the dataset from which validation set will be loaded
+    - username: Hugging Face API username
+    - token: Hugging Face API token
+    """
+    val_ds = load_validation_dataset(dataset_name)
+
+    # model accepts 2D tensors (batch_size, seq_len)
+    val_sequences = torch.tensor([s["tokens"] for s in val_ds])
+
+    if funct_test:
+        val_sequences = val_sequences[:320]
+
+    model = AutoModelForCausalLM.from_pretrained(model_name)
+
+    logprobs_list = []
+    for i in tqdm(range(0, len(val_sequences), batch_size)):
+        batch_sequences = val_sequences[i : i + batch_size]
```

Comment:
Use val_ds here to make a batch tensor. val_ds[tokens] [i:i+b] should work, otherwise with for loop